### ISSUE TITLE
Struggles on iOS 14.4.1 iPhone 12 Pro

### ISSUE THREAD

**Issue by jawang94** [AgoraIO/agora-react-native-rtm#33]
Hey there. I'm using `agora-react-native-rtm` to handle user join subscriptions for audio broadcasting. It works fine on simulator & my iPhone X on iOS 14 (the users list updates & all events are properly fired off). However, on my iPhone 12 Pro running iOS 14.4.1 the `rtmEngine` events fail to fire for seemingly no reason. They simply return a generic Error object w/ no message. `rtcEngine` works properly and I'm able to connect + hear audio coming from the iPhone 12 Pro, so just an `rtmEngine` issue here. Any ideas on what might be going wrong? 

Errors (as you can see I tried logging the `message` and `name` of the first Error object but it doesn't contain any useful info):
<img width="566" alt="Screen Shot 2021-03-17 at 5 20 52 PM" src="https://user-images.githubusercontent.com/41667764/111555206-20ef9f00-8745-11eb-9659-e068c7fd2c71.png">

Initialize method for RTM:
<img width="504" alt="Screen Shot 2021-03-17 at 5 25 54 PM" src="https://user-images.githubusercontent.com/41667764/111555550-dde1fb80-8745-11eb-909c-0833868a2f55.png">

Here's a snippet of the `startCall` method. The first `await` for rtcEngine succeeds but the subsequent 3 `awaits` for rtmEngine fail (only on iPhone 12 Pro running iOS 14.4.1).
<img width="642" alt="Screen Shot 2021-03-17 at 5 22 11 PM" src="https://user-images.githubusercontent.com/41667764/111555292-51373d80-8745-11eb-857d-eb222d91e898.png">

P.S. `agora-react-native-rtm` is on version `^1.2.2-alpha.6` and `react-native-agora` is `^3.3.1`.

Thanks!

**Comment by plutoless** [comment]
is the token null? have you turned on token?

**Comment by t3hfx** [comment]
I have this issue on iPhone 11 14.4 too. I don't send token as soon as my app is in testing mode. 
Just simple "Error" message.

**Comment by jawang94** [comment]
Thanks yeah, I've tried with both a temp token (in testing mode) and a null token. Works on earlier iOS versions but not on newer versions. Again, I've only tested iOS 14.0 and 14.4.1 so I'm not sure at what version this may have started occurring. Is it working for you @plutoless ?

**Comment by RoJoHub** [comment]
> Hey there. I'm using `agora-react-native-rtm` to handle user join subscriptions for audio broadcasting. It works fine on simulator & my iPhone X on iOS 14 (the users list updates & all events are properly fired off). However, on my iPhone 12 Pro running iOS 14.4.1 the `rtmEngine` events fail to fire for seemingly no reason. They simply return a generic Error object w/ no message. `rtcEngine` works properly and I'm able to connect + hear audio coming from the iPhone 12 Pro, so just an `rtmEngine` issue here. Any ideas on what might be going wrong?
> 
> Errors (as you can see I tried logging the `message` and `name` of the first Error object but it doesn't contain any useful info):
> <img alt="Screen Shot 2021-03-17 at 5 20 52 PM" width="566" src="https://user-images.githubusercontent.com/41667764/111555206-20ef9f00-8745-11eb-9659-e068c7fd2c71.png">
> 
> Initialize method for RTM:
> <img alt="Screen Shot 2021-03-17 at 5 25 54 PM" width="504" src="https://user-images.githubusercontent.com/41667764/111555550-dde1fb80-8745-11eb-909c-0833868a2f55.png">
> 
> Here's a snippet of the `startCall` method. The first `await` for rtcEngine succeeds but the subsequent 3 `awaits` for rtmEngine fail (only on iPhone 12 Pro running iOS 14.4.1).
> <img alt="Screen Shot 2021-03-17 at 5 22 11 PM" width="642" src="https://user-images.githubusercontent.com/41667764/111555292-51373d80-8745-11eb-857d-eb222d91e898.png">
> 
> P.S. `agora-react-native-rtm` is on version `^1.2.2-alpha.6` and `react-native-agora` is `^3.3.1`.
> 
> Thanks!

In the next release, we will improve the error code prompt information of our SDK.

Now. please use
```
catch((e)=>{
    console.log(‘code:',e.code);
})
```

-------
Error code For login : [LoginErrorCode](https://docs.agora.io/en/Real-time-Messaging/API%20Reference/RTM_oc/v0.9.3/Constants/AgoraRtmLoginErrorCode.html)

Error code For join channel: [JoinChannelErrorCode](https://docs.agora.io/en/Real-time-Messaging/API%20Reference/RTM_oc/Constants/AgoraRtmJoinChannelErrorCode.html)

Error code For send message[SendMessageErrorCode](https://docs.agora.io/en/Real-time-Messaging/API%20Reference/RTM_oc/Constants/AgoraRtmSendChannelMessageErrorCode.html)

**Comment by jawang94** [comment]
Yeah for whatever reason on the newer iOS versions logging into RTMEngine just returns Error Code 9 for timeout. Doesn't specify why. The other two return Error Code 102 due to not being logged in =/. Any ideas?

**Comment by RoJoHub** [comment]
Maybe you should check you fireware?
https://docs.agora.io/en/Real-time-Messaging/region_java_linux_rtm?platform=Linux#firewall-requirements

**Comment by RoJoHub** [comment]
> I could be wrong but I don't think it has to do with firewalls? This seems to be a problem only with iOS 14.4+. Logging in (both `login()` and `loginV2()`)times out after 6s. Works on iOS 14.0. I'm not really sure what's wrong here. I set up server generated tokens too and I'm using those for both RTC and RTM now. RTC works, but not RTM. I'll keep trying things. Thanks for the help. Hopefully someone else is also having the same problem and can give more info.

Do you mean that iOS can't login by use rtm sdk?
In this case, you should follow this doc.
https://docs.agora.io/en/Real-time-Messaging/messaging_ios?platform=iOS#a-nameappidaget-an-app-id

**Comment by jawang94** [comment]
Yes, correct. This RTM SDK cannot login on iOS 14.4.1. I've tried on two separate iPhones. I'm familiar with the docs and followed the procedures properly. Can you verify that you are able to login to the RTM SDK on iOS version 14.4.1?

**Comment by jawang94** [comment]
Let me give that a try tomorrow! Thanks for the help. Seems to be my issue then for sure.

**Comment by jawang94** [comment]
> > Yes, correct. This RTM SDK cannot login on iOS 14.4.1. I've tried on two separate iPhones. I can confirm it works on earlier iOS version (14.0.0). I'm familiar with the docs and followed the procedures properly. Can you verify that you are able to login to the RTM SDK on iOS version 14.4.1?
> 
> Yes, I confirm that both my iPhone XS Max (14.4.1) and the simulator (14.4.1) will work.
> Maybe you should create project without token on agora console?
> <img alt="WX20210331-140556@2x" width="1314" src="https://user-images.githubusercontent.com/15648309/113097873-417e2580-922a-11eb-82b0-e2c47f3c1498.png">

@RoJoHub It worked for me this morning! I'm not positive what fixed it but these are the changes I made.
1. Update iOS to 14.4.2.
2. Create project without token on Agora console.
3. Completely remove app from device & clean XCode build folder, then rebuild onto device.

Thanks for your help everyone! I'll continue to monitor and try to understand what initially caused the problem.

**Comment by t3hfx** [comment]
Can confirm that creating a project without token worked

**Comment by jawang94** [comment]
Wondering how we might get the token to work since we'd probably want that in production.

**Comment by RoJoHub** [comment]
You may need to consult a backend.

**Comment by t3hfx** [comment]
Yeah, can anybody confirm that they got it working on iOs 14.4 with token?

**Comment by jawang94** [comment]
I have tried it with backend token generation as well as temp tokens via console, neither worked. Would also like to know if someone got it working on iOS 14.4+ w/ token.

**Comment by RoJoHub** [comment]
This involves the authentication mechanism, it is recommended to submit a ticket on the agora official website.

**Comment by RoJoHub** [comment]
Since this does not involve `react-native-agora`, this issue is closed

**Comment by t3hfx** [comment]
Why isn't it linked to agora-react-native-rtm library if it's not working with token only on iPhone 14.4? Android however works

**Comment by jawang94** [comment]
I would like to know that too. I did submit a ticket directly in the Agora console regarding implementation of token based RTM logins so I'll reply here with what they say.

**Comment by jawang94** [comment]
This has been resolved for me although I didn't necessarily do anything differently. Can't explain why it was happening before or how I managed to fix it. May want to leave this open to see if others experience the same issues :)

**Comment by hdcaruth** [comment]
Any update here? I cannot login via token as described above.
